### PR TITLE
Enum symbols are plain strings

### LINKED ISSUE #70: Enum types are not parsed correctly
After packing https://github.com/common-workflow-language/cwl-v1.2/blob/1.2.1_proposed/tests/anon_enum_inside_array_inside_schemadef.cwl using cwlpack and then parsing with cwljava, the enum has symbols like "file:/Users/jdidion/projects/cwlScala/target/test-classes/CommandLineTools/conformance/#anon_enum_inside_array_inside_schemadef.cwl/first/user_type_2/species/homo_sapiens" rather than just "homo_sapiens". Packed workflow below.

```
{
    "cwlVersion": "v1.2",
    "class": "CommandLineTool",
    "requirements": [
        {
            "class": "InlineJavascriptRequirement"
        }
    ],
    "inputs": [
        {
            "type": {
                "name": "user_type_2",
                "type": "record",
                "fields": [
                    {
                        "type": [
                            "null",
                            {
                                "type": "enum",
                                "symbols": [
                                    "homo_sapiens",
                                    "mus_musculus"
                                ]
                            }
                        ],
                        "name": "species"
                    },
                    {
                        "type": [
                            "null",
                            {
                                "type": "enum",
                                "symbols": [
                                    "GRCh37",
                                    "GRCh38",
                                    "GRCm38"
                                ]
                            }
                        ],
                        "name": "ncbi_build"
                    }
                ]
            },
            "id": "first"
        }
    ],
    "baseCommand": "echo",
    "arguments": [
        {
            "prefix": "species",
            "valueFrom": "$(inputs.first.species)"
        },
        {
            "prefix": "ncbi_build",
            "valueFrom": "$(inputs.first.ncbi_build)"
        }
    ],
    "outputs": [
        {
            "type": "stdout",
            "id": "result"
        }
    ],
    "id": "anon_enum_inside_array_inside_schemadef.cwl"
}
```

### PR DESCRIPTION
**Pull request by mr-c**
Use List<String> when we can be more specfic than List<Object>

Implemented using
https://github.com/common-workflow-language/cwl-v1.2/pull/145
and https://github.com/common-workflow-language/schema_salad/pull/507

Fixes #70

### CODE REVIEW

File: src/main/java/org/w3id/cwl/cwl1_2/CommandInputEnumSchemaImpl.java [hunk 1]
```diff
@@ -161,11 +161,11 @@ public CommandInputEnumSchemaImpl(
     } else {
         __baseUri = (String) name.orElse(null);
     }
-    java.util.List<Object> symbols;
+    java.util.List<String> symbols;
     try {
       symbols =
           LoaderInstances
-              .uri_array_of_StringInstance_True_False_None
+              .array_of_StringInstance
```
Comment (mr-c) [inline src/main/java/org/w3id/cwl/cwl1_2/CommandInputEnumSchemaImpl.java:168]: The heart of the fix, part 1

File: src/main/java/org/w3id/cwl/cwl1_2/utils/LoaderInstances.java [hunk 2]
```diff
@@ -113,7 +112,6 @@
   public static Loader<java.util.Optional<java.util.List<Object>>> idmap_fields_optional_array_of_RecordField = new IdMapLoader(optional_array_of_RecordField, "name", "type");
   public static Loader<enum_d9cba076fca539106791a4f46d198c7fcfbdb779> enum_d9cba076fca539106791a4f46d198c7fcfbdb779 = new EnumLoader(enum_d9cba076fca539106791a4f46d198c7fcfbdb779.class);
   public static Loader<enum_d9cba076fca539106791a4f46d198c7fcfbdb779> typedsl_enum_d9cba076fca539106791a4f46d198c7fcfbdb779_2 = new TypeDslLoader(enum_d9cba076fca539106791a4f46d198c7fcfbdb779, 2);
-  public static Loader<java.util.List<Object>> uri_array_of_StringInstance_True_False_None = new UriLoader(array_of_StringInstance, true, false, null);
```
Comment (mr-c) [inline src/main/java/org/w3id/cwl/cwl1_2/utils/LoaderInstances.java:116]: Yuck, good riddance

File: src/test/java/org/w3id/cwl/cwl1_2/utils/AnonymousEnumSymbolNamesTests.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	public void symbolNames() {
+		SchemaDefRequirement schemaDef = (SchemaDefRequirement) this.tool.getRequirements().get().get(0);
+		CommandInputRecordSchema recordDef = (CommandInputRecordSchema) schemaDef.getTypes().get(0);
+		String recordID = recordDef.getName().get();
+		CommandInputRecordField ncbi_build_field = (CommandInputRecordField) recordDef.getFields().get().get(0);
+		Assert.assertEquals(recordID + "/ncbi_build", ncbi_build_field.getName());
+		List<Object> ncbi_build_type = (List<Object>) ncbi_build_field.getType();
+		CommandInputEnumSchema ncbi_build_enum = (CommandInputEnumSchema) ncbi_build_type.get(1);
+		List<String> ncbi_build_symbols = ncbi_build_enum.getSymbols();
+		Assert.assertEquals("GRCh37", ncbi_build_symbols.get(0));
```
Comment (mr-c) [inline src/test/java/org/w3id/cwl/cwl1_2/utils/AnonymousEnumSymbolNamesTests.java:41]: Confirms the fix

**Review by jdidion — APPROVED** [review]

**Comment by mr-c** [comment]
The proposed schema change this was built from [has been rejected by @tetron](https://github.com/common-workflow-language/schema_salad/issues/506#issuecomment-1016875230) ; a new "short name" utility method is the suggested fix: https://github.com/common-workflow-language/schema_salad/issues/511

**Pull request closed without merging**